src/main/java/org/cactoos/scalar/Equality.java

```diff
@@ -62,22 +67,27 @@ public Equality(final T lft, final T rght) {
     public Integer value() throws Exception {
         final byte[] lft = this.left.asBytes();
         final byte[] rght = this.right.asBytes();
+        return new Ternary<>(
+            () -> lft.length == rght.length,
+            () -> Equality.compare(lft, rght),
+            () -> Integer.signum(lft.length - rght.length)
+        ).value();
+    }
+
+    /**
+     * Compare two byte arrays of the same size.
+     * @param lft Left array
+     * @param rght Right array
+     * @return Integer Comparison result
+     */
+    private static int compare(final byte[] lft, final byte[] rght) {
         int result = 0;
-        final int max = Math.max(lft.length, rght.length);
-        for (int idx = 0; idx < max; ++idx) {
-            if (idx >= lft.length) {
-                result = -1;
-                break;
-            }
-            if (idx >= rght.length) {
-                result = 1;
-                break;
-            }
+        for (int idx = rght.length - 1; idx > 0; --idx) {
             result = lft[idx] - rght[idx];
```

Comment:
@Vatavuk If I understand it right, `lft[idx]` will throw `IndexOutOfBoundsException` if `lft` is shorter than `rght`. I think the old logic with `if` checks is more robust. Could you check it, please?

Reply:
@proshin-roman this pice of code is execute only if arrays are equal so there is no fear of exception

Reply:
@proshin-roman Can you provide some argument why should I refactor it in this case?